stdlib/SuiteSparse/src/umfpack.jl

```diff
@@ -558,21 +709,8 @@ end
 
 @inline function getproperty(lu::UmfpackLU, d::Symbol)
     if d === :L || d === :U || d === :p || d === :q || d === :Rs || d === :(:)
-        # Guard the call to umf_extract behaind a branch to avoid infinite recursion
-        L, U, p, q, Rs = umf_extract(lu)
-        if d === :L
-            return L
-        elseif d === :U
-            return U
-        elseif d === :p
-            return p
-        elseif d === :q
-            return q
-        elseif d === :Rs
-            return Rs
-        elseif d === :(:)
-            return (L, U, p, q, Rs)
-        end
+        # Guard the call to umf_extract behind a branch to avoid infinite recursion
+        return umf_extract(lu, d)
```

Comment:
Looking at this, why don't we put all the code in here and have
```julia
umf_extract(lu::UmfpackLU) = getproperty(lu, :(:))
```
for backward compatibility?

Reply:
I doubt anyone has ever used this function since it does nothing more than `getproperty` and it is less convenient. So merging them make sense, moreover  we could get rid of the `if` statement in `getproperty`.